portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -90,11 +106,49 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/blog.html");
   }
 
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+  private static String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     if (value == null) {
       return defaultValue;
     }
     return value;
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private static String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```

Comment:
It's good to avoid null when possible. An alternative is to use Optional<String>.  When there is no upload, returns Optional.empty(). This way, you can change the caller side to  
getUploadedFileUrl(request, "file").ifPresent(fileUrl -> commentEntity.setProperty("fileUrl", fileUrl));